ipfx/bin/run_chirp_fv_extraction.py

```diff
@@ -0,0 +1,176 @@
+                nwb_file=nwb_path, h5_file=h5_path, ontology=ontology)
+        except Exception as detail:
+            logging.warning("Exception when loading specimen {:d} from LIMS".format(specimen_id))
+            logging.warning(detail)
+            return {"error": {"type": "dataset", "details": traceback.format_exc(limit=None)}}
+    elif data_source == "sdk":
+        nwb_path, sweep_info = sdk_nwb_information(specimen_id)
+        try:
+            data_set = AibsDataSet(
+                nwb_file=nwb_path, sweep_info=sweep_info, ontology=ontology)
+        except Exception as detail:
+            logging.warning("Exception when loading specimen {:d} via Allen SDK".format(specimen_id))
+            logging.warning(detail)
+            return {"error": {"type": "dataset", "details": traceback.format_exc(limit=None)}}
+    else:
+        logging.error("invalid data source specified ({})".format(data_source))
+
+
+    # Identify chirp sweeps
+
+    try:
+        iclamp_st = data_set.filtered_sweep_table(clamp_mode=data_set.CURRENT_CLAMP)
+        iclamp_st = data_set.filtered_sweep_table(clamp_mode=data_set.CURRENT_CLAMP, stimuli=["Chirp"])
+        chirp_sweep_numbers = iclamp_st["sweep_number"].sort_values().values
+    except Exception as detail:
+        logging.warning("Exception when identifying sweeps from specimen {:d}".format(specimen_id))
+        logging.warning(detail)
+        return {"error": {"type": "sweep_table", "details": traceback.format_exc(limit=1)}}
+
+    if len(chirp_sweep_numbers) == 0:
+        logging.info("No chirp sweeps for {:d}".format(specimen_id))
+        return {"error": {"type": "processing", "details:": "no available chirp sweeps"}}
+
+    try:
+        result = chirp.extract_chirp_feature_vector(data_set, chirp_sweep_numbers)
+    except Exception as detail:
+        logging.warning("Exception when processing specimen {:d}".format(specimen_id))
+        logging.warning(detail)
+        return {"error": {"type": "processing", "details": traceback.format_exc(limit=1)}}
+
+    return result
+
+
+def run_chirp_feature_vector_extraction(output_dir, output_code, include_failed_cells,
+        specimen_ids, chirp_stimulus_codes, data_source="lims", run_parallel=True):
+    logging.info("Number of specimens to process: {:d}".format(len(specimen_ids)))
+    get_data_partial = partial(data_for_specimen_id,
+                               data_source=data_source,
+                               chirp_stimulus_codes=chirp_stimulus_codes)
+    if run_parallel:
+        pool = Pool()
+        results = pool.map(get_data_partial, specimen_ids)
+    else:
+        results = map(get_data_partial, specimen_ids)
+    filtered_set = [(i, r) for i, r in zip(specimen_ids, results) if not "error" in r.keys()]
+    error_set = [{"id": i, "error": d} for i, d in zip(specimen_ids, results) if "error" in d.keys()]
+    if len(filtered_set) == 0:
+        logging.info("No specimens had results")
+        return
```

Comment:
This block of code is repeated from run_feature_vector_extraction.py. Could you consider putting it into a separate function?

Reply:
Yes, I can refactor some common things from the runner scripts into a `bin_utils` module or something like that.